utils/esbuild-plugin-cjs-interop.ts

```diff
@@ -0,0 +1,54 @@
+import path from 'path';
+
+/**
+ * "Fixes" the esbuild problem of exporting the CJS default export as `module.exports.default`
+ * instead of `module.exports`. The plugin appends a block of code to the file that takes the
+ * `.default` module export and re-exports it as `module.exports`. It then takes all named
+ * exports and re-exports them as part of the `module.exports` under the same name. This also
+ * gives the benefit of exporting the `.default` module which allows us to support all 3 export
+ * styles: the default export, `.default` export, and named exports.
```

Comment:
Is there an issue with ESBuild open on this? If not, could you submit one? This feels like it would be a generally useful feature.